### PR TITLE
feat: Handle Fn::If intrinsic for Alarms

### PR DESCRIPTION
**Pull request by mgrandis**
**Issue #, if available:**
#1624 

**Description of changes:**
Handles `Fn::If` in `DeploymentPreference/Alarms` where we would only allow a list before.

**Description of how you validated changes:**
- Unit tests
- E2E tests
- Deployed the resulting CFN template and verified the stack content
- Integration test

**Checklist:**
- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected

**Examples?**

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/preferences/deployment_preference_collection.py [hunk 1]
```diff
@@ -152,6 +152,67 @@ def deployment_group(self, function_logical_id):
...
+        group : dict
+            Deployment group
+
+        Raises
+        ------
+        ValueError
+            If Alarms is in the wrong format
+        """
+        if not preference.alarms or is_intrinsic_no_value(preference.alarms):
+            return
```
Comment (elbayaaa) [inline samtranslator/model/preferences/deployment_preference_collection.py:172]: what if it is in the following of: [Fn::If, condition, AWS::NoValue, someValue] and the condition may or maynot evaluates to true?

Comment (mgrandis) [reply samtranslator/model/preferences/deployment_preference_collection.py:172]: We won't enter this condition and proceed to line 174

File: samtranslator/model/preferences/deployment_preference_collection.py [hunk 2]
```diff
@@ -152,6 +152,67 @@ def deployment_group(self, function_logical_id):
 
         return deployment_group
 
+    def _process_alarms(self, preference, group):
```
Comment (elbayaaa) [inline samtranslator/model/preferences/deployment_preference_collection.py:155]: I see "group" is used only as an output parameter(lines 181,  & 184) I would prefer to remove it and make the call that way:
```
group.AlarmConfiguration = self._process_alarms(deployment_preference);
```

Comment (mgrandis) [reply samtranslator/model/preferences/deployment_preference_collection.py:155]: Indeed and return None at line 172

Comment (elbayaaa) [reply samtranslator/model/preferences/deployment_preference_collection.py:155]: in this case, you might want to rename "process_alarms" to something more descriptive

**Review by elbayaaa — APPROVED** [review]

File: samtranslator/model/preferences/deployment_preference_collection.py [hunk 3]
```diff
@@ -152,6 +152,69 @@ def deployment_group(self, function_logical_id):
...
+            If Alarms is in the wrong format
+        """
+        if not preference_alarms or is_intrinsic_no_value(preference_alarms):
+            return None
+
+        if is_intrinsic_if(preference_alarms):
+            processed_alarms = copy.deepcopy(preference_alarms)
+            alarms_list = processed_alarms.get("Fn::If")
+            if not isinstance(alarms_list, list) or not len(alarms_list) == 3:
+                raise ValueError("Fn::If requires 3 arguments")
```
Comment (hawflau) [inline samtranslator/model/preferences/deployment_preference_collection.py:181]: nit: it looks like a generic check on `Fn::If`. I think it's better to refactor this part to intrinsics.py

**Review by hawflau — APPROVED** [review]

**Pull request merged**